src/dlsnode/storage/util/InputBuffer.d

```diff
@@ -77,6 +102,25 @@ struct InputBuffer
         this.position_in_chunk += dest.length;
     }
 
+    /**************************************************************************
+
+        Consumes required.length bytes from the internal buffer and sinks them
+        into the `sink` delegate keeping the track of the position in the
+        internal buffer.
+
+        Params:
+            required = number of bytes required to consume
+            sink = sink delegate to fill from the internal buffer.
+
+    **************************************************************************/
+
+    private void consumeFromInternalBuffer (size_t required, void delegate(in void[] dest) sink)
+    {
```

Comment:
Verify that the buffer contains at least the required number of bytes?

Reply:
Ah, good suggestion.